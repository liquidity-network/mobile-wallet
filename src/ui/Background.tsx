import React, { PureComponent } from 'react'
import { Image, StyleSheet, ImageProps } from 'react-native'

import { deviceHeight, deviceWidth } from 'globalStyles'

export default class Background extends PureComponent<ImageProps> {
  render() {
    return <Image {...this.props} style={imageStyle} />
  }
}

const imageStyle: any = {
  ...StyleSheet.absoluteFillObject,
  width: deviceWidth,
  height: deviceHeight,
  resizeMode: 'cover',
}
