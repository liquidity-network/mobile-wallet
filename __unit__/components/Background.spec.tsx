import React from 'react'
import { create } from 'react-test-renderer'

import Background from 'ui/Background'

describe('<Background> component', () => {
  it('renders correctly', () => {
    const tree = create(<Background source={{ uri: 'https://google.com' }} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
