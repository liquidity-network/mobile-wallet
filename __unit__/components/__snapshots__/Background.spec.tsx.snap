// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background> component renders correctly 1`] = `
<Image
  source={
    Object {
      "uri": "https://google.com",
    }
  }
  style={
    Object {
      "bottom": 0,
      "height": 1334,
      "left": 0,
      "position": "absolute",
      "resizeMode": "cover",
      "right": 0,
      "top": 0,
      "width": 750,
    }
  }
/>
`;
